### ISSUE TITLE
Episodes doesn't appear unless a specific type of naming format is already used.

### ISSUE THREAD

**Issue by 5abred** [Komet/MediaElch#1356]
**Describe the bug**
when importing a TV show if the files inside the tv show folder aren't already named S<XX>.E<XX> format, then the episode won't get imported under the Show name.

**To Reproduce**
 if I add a source where my tv show is residing and the episodes are named (Episode1, Episode 2 ..etc) or (1,2,3...) the Series doesn't show the available episodes in the folder I'll need to rename them according to this format and this format only in order to appear.

**Expected behavior**
It should read the numbering, present the available episodes regardless of the current numbering format, and then rename them according to my needed format later.

**Screenshots**

**MediaElch Version:**
2.8.12 "Coridian"

**Operating System:**
Windows

**Comment by ticao2** [comment]
I'm not sure but I think MediaElch follows Kodi's rules.  
https://kodi.wiki/view/Naming_video_files/Episodes  
I don't know if MediaElch supports ALL formats that Kodi supports.  
@bugwelle  can clarify better.

**Comment by 5abred** [comment]
If that's the case then this should be changed to feature request, since it's would be convenient for MediaElch to read numbering of any kind and template, for example not every download is gonna follow kodi template, it migh be named (title {episode #}

**Comment by rassaei** [comment]
I am not 100% sure, but I think I have a similar problem. But Mediaelch also doesn't find my Episodes when I rename them. They just do not appear.
The debug log also sheds no light on this unfortunately...
This might be a related issue: https://github.com/Komet/MediaElch/issues/1028

**Comment by bugwelle** [comment]
Hi there, 

sorry for the _really_ late reply, but I was writing my Master thesis for the past few months and didn't have any time for MediaElch. :)

The issue is that MediaElch can't check for all numbers as that would lead to false positives. Many users encode the video resolutionn in the file name, etc.

But I could extend the current implementation to also cover different naming conventions. For that I'd need the file names that you used.

An "import episode" feature would definitely be useful, but I don't have time for that in the near future. :)


Regards,
Andre

**Comment by rassaei** [comment]
First of all: I hope the thesis went well :)

I think I didn't explain the issue sufficiently.
If I rename the file to "s01e01.mp4" and place it into "tvshows/sampleshow" then MediaElch finds "Sampleshow" but shows 0 new episodes.
If I do the same with a file that MediaElch finds in other places and with other names, it works without any issues.

Is there anything I could provide, so you could replicate the issue?

**Comment by ticao2** [comment]
Maybe it's not a problem to be able to read the episode filename.  
I think the Season folder is missing.  
`TV Shows
	SampleShow01
		Season01
			s01e01.mp4
			s01e02.mp4
			s01e03.mp4
			s01e04.mp4`

**Comment by rassaei** [comment]
No.
As I said, with other videofiles it works. It has nothing to do with the filename or the directories.
I don't understand this issue as the debug log is the same if I use files that are found or file that are not.

Mediaelch finds videos when they are in the directory of the show. New files don't need to be in a season directory.

**Comment by bugwelle** [comment]
> First of all: I hope the thesis went well :)

Yes, thanks for asking :smile: 

> I think I didn't explain the issue sufficiently. If I rename the file to "s01e01.mp4" and place it into "tvshows/sampleshow" then MediaElch finds "Sampleshow" but shows 0 new episodes. If I do the same with a file that MediaElch finds in other places and with other names, it works without any issues.
> 
> Is there anything I could provide, so you could replicate the issue?

Thanks for your answer. Do you have an `advancedsettings.xml` file? If so, could you paste the contents here? If not, that's fine as well.

Could you paste the contents of the Developer information here? See <https://mediaelch.github.io/mediaelch-doc/contributing/bug-reports.html#developer-information>

And could you still describe your directory layout? It may not seem like it, but it may be a reason nothing shows up.

> Mediaelch finds videos when they are in the directory of the show. New files don't need to be in a season directory.

I'm not sure that's intended and will have to check. :)


Regards,
Andre

**Comment by mdkberry** [comment]
old issue  but I was running into this with the latest version on windows.

I found if I add the year in brackets then it finds them. 

I have things like _"\Sid Vicious - Various Videos"_ in a _"\Docuseries"_ folder and MediaElch was not picking up some of the videos inside it when the top folder is treated as tv show folder. 

When I added the year into the name it seems to be happy then:

e.g.  _"Deaths of Sid and Nancy Leave Many Unanswered Questions, 40 Years Later [Z9tJUvI6uHQ].mp4"_
change the name to _"Deaths of Sid and Nancy Leave Many Unanswered Questions, 40 Years Later (2018) [Z9tJUvI6uHQ].mp4"_
and it seems to find them
